operator/api/v1alpha1/condition_provider.go

```diff
@@ -0,0 +1,54 @@
+package v1alpha1
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SaveStatus will trigger an object update to save the current status
+// conditions
+func (httpso *HTTPScaledObject) SaveStatus(
+	logger logr.Logger,
+	cl client.Client,
+) {
+	ctx := context.TODO()
```

Comment:
@khaosdoctor can you have `SaveStatus` take in a context instead of this?

Reply:
Yes it's possible, I'll do it